File: backend/test_db_connection.py
#!/usr/bin/env python3
"""
Test script to verify database connection and session creation
"""
import sys
import os
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

from config import settings
from database.postgres import engine, Base
from services.session_service import session_service

def test_db_connection():
    print(f"Database URL: {settings.database_url}")

    try:
        # Test basic connection
        from sqlalchemy import text
        with engine.connect() as conn:
            result = conn.execute(text("SELECT 1"))
            print("[OK] Basic database connection successful")
    except Exception as e:
        print(f"[ERROR] Basic database connection failed: {e}")
        return False

    try:
        # Create tables
        Base.metadata.create_all(bind=engine)
        print("[OK] Database tables created successfully")
    except Exception as e:
        print(f"[ERROR] Table creation failed: {e}")
        return False

    try:
        # Test session creation
        session_id = session_service.create_session()
        if session_id:
            print(f"[OK] Session created successfully: {session_id}")
        else:
            print("[ERROR] Session creation failed - returned empty ID")
            return False
    except Exception as e:
        print(f"[ERROR] Session creation failed: {e}")
        return False

    return True

if __name__ == "__main__":
    print("Testing database connection and session creation...")
    success = test_db_connection()
    if success:
        print("\n[SUCCESS] All tests passed!")
    else:
        print("\n[FAILURE] Some tests failed!")
        sys.exit(1)

File: my-website/test-db-connection.js
import Database from 'better-sqlite3';
import dotenv from 'dotenv';

dotenv.config();

console.log('Testing database connection...');
console.log('DATABASE_URL from env:', process.env.DATABASE_URL);

const dbPath = process.env.DATABASE_URL || './db.sqlite';
console.log('Using database path:', dbPath);

try {
  const db = new Database(dbPath);
  console.log('Database connection successful');

  // Test basic operations
  const result = db.prepare('SELECT sqlite_version() as version').get();
  console.log('SQLite version:', result.version);

  // Create a simple table to test write operations
  db.exec('CREATE TABLE IF NOT EXISTS test_table (id INTEGER PRIMARY KEY, name TEXT)');
  console.log('Test table created successfully');

  // Insert a test record
  const insert = db.prepare('INSERT INTO test_table (name) VALUES (?)');
  const info = insert.run('test_record');
  console.log('Test record inserted with id:', info.lastInsertRowid);

  // Read the test record
  const select = db.prepare('SELECT * FROM test_table WHERE id = ?').get(info.lastInsertRowid);
  console.log('Retrieved test record:', select);

  db.close();
  console.log('Database test completed successfully');
} catch (error) {
  console.error('Database test failed:', error.message);
}